webapp/blog/views.py

```diff
@@ -195,22 +195,24 @@ def snap_posts(snap):
     articles = []
 
     if blog_tags:
-        blog_tags_ids = logic.get_tag_id_list(blog_tags)
-        try:
-            blog_articles, total_pages = api.get_articles(blog_tags_ids, 3)
-        except ApiError:
-            blog_articles = []
-
-        for article in blog_articles:
-            transformed_article = logic.transform_article(
-                article, featured_image=None, author=None
-            )
-            articles.append(
-                {
-                    "slug": transformed_article["slug"],
-                    "title": transformed_article["title"]["rendered"],
-                }
-            )
+        blog_tags_ids = logic.get_tag_id_list(blog_tags, snap)
+
+        if blog_tags_ids:
```

Comment:
I don't think this will ever be empty if blog_tags is not empty because if it can't get the id it will just raise a `KeyError`.

Reply:
`get_tag_id_list` returns an empty list if tags are found, but don't match the required name.

So the next part of the code would run (without any tag filtering) and return the latest 3 posts from the blog either way

Reply:
You are absolutely right, sorry didn't read the `get_tag_id_list` function properly!